vagrant/provisioning/roles/arkcase-portal-ui/tasks/main.yml

```diff
@@ -45,3 +45,19 @@
     src: "{{ root_folder }}/install/httpd/arkcase-portal{{ arkcase_portal_extension_formatted }}-ui{{ arkcase_portal_ui_version_formatted }}.zip"
     dest: "{{ root_folder }}/data/httpd/htdocs/{{ arkcase_portal_context }}/"
   when: arkcase_portal_zip is changed
+
+- name: see if arkcase portal UI robots.txt exists
+  become: yes
+  become_user: apache
+  stat:
+    path: "{{ root_folder }}/data/httpd/htdocs/{{ arkcase_portal_context }}"
```

Comment:
It looks like you're checking whether a folder exists, and not whether robots.txt appears in the folder.